File: cplusplus/design_pattern/AdapterProj/Adapter2.cpp
#include "stdafx.h"
#include "Adapter2.h"


Adapter2::Adapter2()
{
}


Adapter2::~Adapter2()
{
}


void Adapter2::sayHi()
{
    myAdaptee.sayHi();
}


void Adapter2::sayHello()
{
    std::cout << "void Adapter2::sayHello()" << std::endl;
}


Adapter2::Adapter2(const AdapterObject & adaptee):myAdaptee(adaptee)
{
}

File: Java/pattern/adapter_pattern/src/com/pattern/Adapter2.java
package com.pattern;

public class Adapter2 implements TargetClass {
	AdapteeObject myAdaptee;

	public Adapter2(AdapteeObject myAdaptee) {
		super();
		this.myAdaptee = myAdaptee;
	}

	@Override
	public void sayHi() {
		myAdaptee.sayHi();
	}

	@Override
	public void sayHello() {
		System.out.println("Hello");
	}

}
